src/core/tree.ml

```diff
@@ -256,7 +252,9 @@ module CM = struct
     (** Index of next slot to use in [vars] array to store variables. *)
     ; positions : arg list
     (** Positions of the elements of the matrix in the initial term. Note that
-        we rely on the order relation used in sets. *) }
+        we rely on the order relation used in sets. *)
+    ; rule_order: bool
+    (** TODO *) }
```

Comment:
Why do you need to change anything here? Isn't it sufficient that a boolean argument is added to compile (and propagated wherever it is necessary)?

Reply:
This is precisely the propagation: to know that we are building the tree of a symbol which keeps rule order.

Reply:
Ah I see what you mean, I'll have a look, but it looks like your idea is better.